atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/Ec2HostLocationSupplier.java

```diff
@@ -35,24 +36,18 @@
  */
 public final class Ec2HostLocationSupplier implements Supplier<HostLocation> {
     private static final Logger log = LoggerFactory.getLogger(Ec2HostLocationSupplier.class);
-
-    /*
-       This is a supplier to avoid class loading races breaking downstream internal products.
-    */
-    private static final Supplier<OkHttpClient> client = Suppliers.memoize(() -> new OkHttpClient.Builder().build());
+    private static final String AZ_URL = "http://169.254.169.254/latest/meta-data/placement/availability-zone";
 
     @Override
     public HostLocation get() {
         try {
-            Response response = client.get()
-                    .newCall(new Request.Builder()
-                            .get()
-                            .url("http://169.254.169.254/latest/meta-data/placement/availability-zone")
-                            .build())
-                    .execute();
-            Preconditions.checkState(response.isSuccessful(), "Getting AWS host metadata was not successful");
-
-            return parseHostLocation(response.body().string());
+            URL url = new URL(AZ_URL);
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            int responseCode = conn.getResponseCode();
+            Preconditions.checkState(responseCode == 200, "Getting AWS host metadata was not successful");
+            try (InputStreamReader reader = new InputStreamReader(conn.getInputStream())) {
+                return parseHostLocation(CharStreams.toString(reader));
+            }
```

Comment:
Does this need to disconnect the connection?

```suggestion
            try {
                int responseCode = conn.getResponseCode();
                Preconditions.checkState(responseCode == 200, "Getting AWS host metadata was not successful");
                try (InputStreamReader reader = new InputStreamReader(conn.getInputStream())) {
                    return parseHostLocation(CharStreams.toString(reader));
                }
            } finally {
                conn.disconnect();
            }
```

Reply:
It's not necessary for it to, so I figured it'd keep things simple by not.